crates/bevy_render/src/renderer/graph_runner.rs

```diff
@@ -58,18 +58,12 @@ impl RenderGraphRunner {
         queue: &wgpu::Queue,
         world: &World,
     ) -> Result<(), RenderGraphRunnerError> {
-        let command_encoder =
-            render_device.create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
-        let mut render_context = RenderContext {
-            render_device,
-            command_encoder,
-        };
-
+        let mut render_context = RenderContext::new(render_device);
         Self::run_graph(graph, None, &mut render_context, world, &[])?;
         {
             #[cfg(feature = "trace")]
             let _span = info_span!("submit_graph_commands").entered();
-            queue.submit(vec![render_context.command_encoder.finish()]);
+            queue.submit(render_context.finish());
```

Comment:
Is submitting all the render bundles at once a good solution? Doesn't that mean, that all GPU render operations only start executing at the end of the render app schedule, thus increasing latency and even frame times?
I always thought, that submitting work as soon as possible is beneficial, to keep the GPU occupied. Is this assumption correct?

Could we add the `Queue` to the render context, so that we can decide when to submit work inside the graph? 

Also related to the question above: If submitting as soon as possible is beneficial, then shouldn't we move as much logic as possible into the render graph, instead of running it beforehand in systems? If I understand this solution correctly, then we would record the buffers beforehand and only execute the graph once all of them have finished, even though we could already be submitting some buffers to occupy the GPU.

Reply:
We can flush them as they come in, in `add_command_buffer` if need be. Though that probably will result in more submit calls, which may or may not have higher overhead. Worth testing.

Note this is currently not a departure from what we currently already are doing. We're deferring the singular command buffer from being submitted until after the render graph is done running. The aim here is to reduce the CPU-side burden by parallelizing the command buffer encoding across threads. How we collect those buffers/bundles into a coherent rendered frame can still be changed after this.